docs/python-reference/dtype.ipynb

```diff
@@ -82,8 +91,163 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "Scipp supports common dtypes like\n",
+    "- `float32`, `float64`\n",
+    "- `int32`, `int64`\n",
+    "- `bool`\n",
+    "- `string`\n",
+    "- `datetime64`\n",
+    "\n",
+    "You can get a full list using \n",
+    "```python\n",
+    "[s for s in dir(sc.dtype) if not s.startswith('__')]\n",
+    "```\n",
+    "but note that many of those dtypes are only meant for internal use.\n",
+    "\n",
     "[Customizing Scipp](../developer/customizing.rst) explains how support for additional types can be added."
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dates and Times\n",
+    "\n",
+    "Scipp has a special dtype for time-points, `sc.dtype.datetime64`.\n",
+    "Variables can be constructed from integers which encode the time since the [Unix epoch](https://en.wikipedia.org/wiki/Unix_time):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc.scalar(value=0, unit=sc.units.s, dtype=sc.dtype.datetime64)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc.scalar(value=681794055, unit=sc.units.s, dtype=sc.dtype.datetime64)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Datetime Variables always need a temporal unit and that unit determines how the integer that is passed to `value=` is interpreted:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "var = sc.scalar(value=681794055, unit=sc.units.ns, dtype=sc.dtype.datetime64)\n",
```

Comment:
Is it relevant to say (or show) what happens if a non-time unit is used?

Reply:
Not sure. I generally prefer avoiding showing how not to do things. Is it useful to know that you can swap out the unit and effectively break the variable?